### DISCUSSION TITLE
Image with Keystatic

### DISCUSSION THREAD

**Discussion by emilienbidet** [Thinkmill/keystatic#427] [Ideas]
Hello Keystatic lovers

I'm currently building a hotel directory and I played a little bit with the image field. Here are my feedbacks :

## Array item image preview

<img width="550" alt="CleanShot 2023-07-06 at 20 35 59@2x" src="https://github.com/Thinkmill/keystatic/assets/44983373/8a6bb22a-b61d-4d8c-8ece-6d4259148805">

It would be super cool for array objects to add a item preview image !
We could simply do the same as `itemLabel` and add `itemPreview` field with the path of the picture.

## Custom image filename

For SEO purpose I need that my image filename to be customisable.
I don't have any idea of implementation but I'm super open to a solution.

<img width="480" alt="CleanShot 2023-07-06 at 20 45 48@2x" src="https://github.com/Thinkmill/keystatic/assets/44983373/876d36da-46b8-405e-9bbd-740e6d6cc574">

Because today the url is pretty bad for SEO :)

Thank you everyone !
Emilien

**Comment by JedWatson** [comment]
Thanks for the feedback @emilienbidet!

Both good points, we'll have a look and see what we can do 👍

**Reply by emilienbidet** [reply]
Thank you @JedWatson
